spec/models/upload_spec.rb

```diff
@@ -229,6 +229,59 @@
     end
   end
 
+  context ".get_from_url" do
+    let(:upload) { Fabricate(:upload, sha1: "10f73034616a796dfd70177dc54b6def44c4ba6f") }
+    let(:upload2) { Fabricate(:upload, sha1: "2a7081e615f9075befd87a9a6d273935c0262cd5") }
+
+    it "works with multiple uploads" do
+      expect(Upload.get_from_urls([upload.url, upload2.url])).to contain_exactly(upload, upload2)
+    end
+
+    it "works when the file has been uploaded" do
+      expect(Upload.get_from_urls([upload.url])).to contain_exactly(upload)
+    end
+
+    it "works for an extensionless URL" do
+      upload.update!(url: upload.url.sub('.png', ''))
+      expect(Upload.get_from_urls([upload.reload.url])).to contain_exactly(upload)
```

Comment:
Feels like these two lines are doing too much. It would be clearer with more code.

```suggestion
      url = upload.url.sub('.png', '')
      upload.update!(url: url)
      upload.reload
      expect(Upload.get_from_urls([url])).to contain_exactly(upload)
```